kernel/kernel-impl/src/main/java/org/sakaiproject/authz/impl/BaseAuthzGroupService.java

```diff
@@ -1101,6 +1105,16 @@ public void refreshUser(String userId)
 			}
 
 			siteService.setUserSecurity(userId, updSites, unpSites, visitSites);
+			
+			// Get the sites in which this user has been added or removed
+			Collection changedVisitSites = CollectionUtils.disjunction(oldVisitAuthzGroups, visitAuthzGroups);
+			for (Iterator i = changedVisitSites.iterator(); i.hasNext();) {
+				String azGroupId = (String) i.next();
+				Reference ref = entityManager().newReference(azGroupId);
+				 
+				SiteParticipantNotification.notifySiteParticipant("/gradebook/" + ref.getId() + "/");
```

Comment:
Having said that there already looks to be direct binding to the gradebook in the conditions kernel code 😞